### PR TITLE
Make test pass standalone

### PR DESCRIPTION
**Pull request by lacostej**
### Checklist

- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I see several green `ci/circleci` builds in the "All checks have passed" section of my PR ([connect CircleCI to GitHub](https://support.circleci.com/hc/en-us/articles/360008097173-Why-aren-t-pull-requests-triggering-jobs-on-my-organization-) if not)
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.
- [x] I've added or updated relevant unit tests.

### Motivation and Context

Some people's tests fail on their dev env.

### CODE REVIEW

**Review by taquitos — APPROVED** [review]
Thank you!
:shipit:

**Comment by lacostej** [comment]
Not really sure why this error comes now

```
/Applications/Xcode-15.2.0.app/Contents/Developer/usr/bin/xcodebuild -resolvePackageDependencies -workspace gym/spec/fixtures/projects/projectBasedOnInheritance/ExampleWithInheritedXcconfig.xcworkspace -scheme "Target B" build
Command line invocation:
    /Applications/Xcode-15.2.0.app/Contents/Developer/usr/bin/xcodebuild -resolvePackageDependencies -workspace gym/spec/fixtures/projects/projectBasedOnInheritance/ExampleWithInheritedXcconfig.xcworkspace -scheme "Target A" build

User defaults from command line:
    IDEPackageSupportUseBuiltinSCM = YES

2024-10-01 07:40:36.033 xcodebuild[56563:52022096] Writing error result bundle to /var/folders/3n/52m9bkhs7fl5mzc46rvdk2dh0000gp/T/ResultBundle_2024-01-10_07-40-0036.xcresult
xcodebuild: error: Scheme Target A is not currently configured for the build action.
```

File: gym/spec/code_signing_mapping_spec.rb [hunk 1]
```diff
@@ -28,38 +28,40 @@
...
 
     it "detects the build configuration from selected scheme of a project based on inheritance for resolve xcconfigs", requires_xcode: true do
       workspace_path = "gym/spec/fixtures/projects/projectBasedOnInheritance/ExampleWithInheritedXcconfig.xcworkspace"
-      project = FastlaneCore::Project.new({ workspace: workspace_path })
+      options = { workspace: workspace_path, scheme: "Target A" }
+      project = FastlaneCore::Project.new(options)
+      Gym.config = FastlaneCore::Configuration.create(Gym::Options.available_options, options)
       csm = Gym::CodeSigningMapping.new(project: project)
-      Gym.config[:scheme] = "Target A"
+      #Gym.config[:scheme] = "Target A"
```
Comment (taquitos) [inline gym/spec/code_signing_mapping_spec.rb:53]: `#`

File: spaceship/spec/portal/provisioning_profile_template_spec.rb [hunk 2]
```diff
@@ -1,4 +1,5 @@
 describe Spaceship::Portal::ProvisioningProfileTemplate do
+  before { Spaceship.login }
```
Comment (taquitos) [inline spaceship/spec/portal/provisioning_profile_template_spec.rb:2]: Huh, this is unexpected, but I trust you 😅

File: deliver/spec/sync_screenshots_spec.rb [hunk 3]
```diff
@@ -1,10 +1,8 @@
 require 'deliver/sync_screenshots'
 require 'fakefs/spec_helpers'
-require_relative 'deliver_constants'
```
Comment (taquitos) [inline deliver/spec/sync_screenshots_spec.rb:3]: Nice!

**Review by taquitos — APPROVED** [review]
Really great investigative work here. 🚀  it

File: spaceship/spec/portal/provisioning_profile_template_spec.rb (hunk 2 above)
Comment (lacostej) [reply spaceship/spec/portal/provisioning_profile_template_spec.rb:2]: image

**Pull request merged**